### ISSUE TITLE
License vim-template under a free software license?

### ISSUE THREAD

**Issue by ArniDagur** [aperezdc/vim-template#128]
Athough this project is open source, it technically is not free software as defined by the FSF as it has not been licensed under such a license. This is important as there are many people who refuse to use any non-free software.

What is this project's licensing stance? Have you simply not gotten around to adding a license, or do you intend to reserve all rights to the software? Clarification would be appreciated.

**Comment by aperezdc** [comment]
At least the main code of the plugin is clearly [MIT-licensed as per the file header](https://github.com/aperezdc/vim-template/blob/master/plugin/templates.vim#L7), but I do agree with @ArniDagur that it would be better to have a clearer licensing, with a `COPYING` text file added to the root of the repository.

To make things as transparent and clear as possible, I have assembled the following list of the rest of the files, and the people who contributed to them. If you are in the list, please reply to this issue explicitly telling whether you agree on licensing those under the terms of the [MIT license](https://opensource.org/licenses/MIT):

- [ ] [https://github.com/aperezdc/vim-template/blob/master/README.rst](README.rst)
  - [ ] @Lyude 
  - [ ] @muff1nman 
  - [ ] @FacundoAcevedo 
  - [ ] @infmagic2047 
  - [ ] @Dewdrops 
  - [ ] @chenjianlong 
  - [ ] @bradrice 
- [ ] [syntax/vim-template.vim](https://github.com/aperezdc/vim-template/blob/master/syntax/vim-template.vim)
  - [ ] @Lyude
  - [ ] @bstin 
- [ ] [doc/template.txt](https://github.com/aperezdc/vim-template/blob/master/doc/template.txt)
  - [ ] @muff1nman 
  - [ ] @ArniDagur 
  - [ ] @toransahu 
  - [ ] @pgraham

**Comment by aperezdc** [comment]
For the example template files under the [templates/](https://github.com/aperezdc/vim-template/tree/master/templates) subdirectory, I think it makes sense to add an exception to the license, saying that any derived works do not need to follow the MIT license. The reason for this is that the text that they generate when the templates are expanded may end up being part of any project, with any license; if we don't add the exception then technically would force projects using templates to have a license that is MIT-compatible... and I think that would be quite bad.

The following people have contributed to the files under `templates/`; please comment here in the next days if you disagree with the proposed license exception:

- [ ] @ogarcia 
- [ ] @muff1nman 
- [ ] @muse 
- [ ] @timhughes 
- [ ] @rharriso 
- [ ] @disoul 
- [ ] @olshevskiy87
- [ ] @dimkarakostas 
- [ ] @libaoxi
- [ ] Jon Bartlet (note: I'll try to send him an e-mail).

**Comment by FacundoAcevedo** [comment]
I do prefer GPLv3, but since it has been released as MIT, I will not block consensus, so:

I agree on licensing my contribution under the terms of the MIT license.

**Comment by bstin** [comment]
I agree. the MIT license is fine with me.

**Comment by ArniDagur** [comment]
> For the example template files under the templates/ subdirectory, I think it makes sense to add an exception to the license, saying that any derived works do not need to follow the MIT license. The reason for this is that the text that they generate when the templates are expanded may end up being part of any project, with any license; if we don't add the exception then technically would force projects using templates to have a license that is MIT-compatible... and I think that would be quite bad.

I have three thoughts on this:

1. There is this concept in copyright law known as the [threshold of originality](https://en.wikipedia.org/wiki/Threshold_of_originality). Basically, since the templates are all made up of boilerplate code and simple, unoriginal (as in not being an original work) sentances, they do not reach the 'threshold of originality', and are thus not protected by copyright; they are in the public domain.

2. Even in a scenario where the templates hold copyright protection, licensing them under a permissive license such as MIT would be preferable to all rights being reserved.

3. Explicitly waiving all copyright by—for example—using the Unlicense, would be even better.

---

**On another note:**

I agree on licensing my contributions to this project under the terms of the MIT license.

**Comment by olshevskiy87** [comment]
hi! MIT is ok for me

**Comment by aperezdc** [comment]
While I would be fine using another license and I do like the GPL myself, but given that the MIT license was chosen back when first published the plugin (maybe an oversight, thinking in retrospective), the path of least resistance is keeping the same license.

Regarding the `templates/` subdir, I also think that @ArniDagur's idea of using the [Unlicense](https://unlicense.org/) is a pretty good idea.

@ogarcia From your comment I understand you would agree to using the Unlicense for the content under `templates/`. Could you please explicitly confirm?

@olshevskiy87 Sorry for pinging you again. Could you please also confirm whether using the Unlicense for content under `templates/` is okay for you?

**Comment by ogarcia** [comment]
@aperezdc I confirm :+1:

**Comment by aperezdc** [comment]
Also, while th gathering of consensus is in process, let's not merge any PR that would affect files which do not have explicit licensing terms. That means that I will only merge commits that touch `plugin/templates.vim` in the meantime.

**Comment by dimkarakostas** [comment]
@aperezdc I also agree with using the Unlicense (or MIT for that matter) for the content under `templates/`.

**Comment by toransahu** [comment]
@aperezdc  I agree licensing it under the terms of [MIT license](https://opensource.org/licenses/MIT).

**Comment by olshevskiy87** [comment]
@aperezdc 

> Could you please also confirm whether using the Unlicense for content under templates/ is okay for you?

confirm

**Comment by pgraham** [comment]
I agree to licensing under the terms of the MIT license.

**Comment by rharriso** [comment]
I agree

**Comment by timhughes** [comment]
I agree for Unlicense for templates/

**Comment by muff1nman** [comment]
Sorry for not noticing this sooner. I agree to have source relicensed under the MIT license and the templates to be unlicensed.

**Comment by muse** [comment]
👍

**Comment by chenjianlong** [comment]
I agree

**Comment by infmagic2047** [comment]
Sorry for the late reply. It's OK for me. :)

**Comment by libaoxi** [comment]
Sorry i am late, I agree.
